### ISSUE TITLE
about the default value of min_lr

### ISSUE THREAD

**Issue by xu-song** [pytorch/fairseq#2523]
#### What is your question?

 I am confused about that the default value of `min_lr` is `-1`. 

When the `min_lr` reaches 0, training is still  in progress.

```
lr=[0.0005], lr_scheduler='polynomial_decay'
```
#### Code

https://github.com/pytorch/fairseq/blob/226f0e45391ac1dcacb72ff68b523bf7b2ebceda/fairseq/options.py#L476

https://github.com/pytorch/fairseq/blob/226f0e45391ac1dcacb72ff68b523bf7b2ebceda/fairseq_cli/train.py#L117-L120

## Another question

How can I resume training when `min_lr` reach 0? How to increase lr to a positive value.

**Comment by alexeib** [comment]
you can update your max-updates (and --total-num-update), or use --reset-lr-scheduler and specify some different learning rate scheduler

min lr is initialized to -1 because in some cases you start off with a learning rate of 0 (like if you are warming up from 0 to some number) and dont want the training to stop immediately
